src/pages/non-pass-finalisation/cat-a-mod1/non-pass-finalisation.cat-a-mod1.page.ts

```diff
@@ -161,6 +162,14 @@ export class NonPassFinalisationCatAMod1Page extends BasePageComponent implement
         select(isWelshTest),
       ),
     };
+
+    currentTest$.pipe(
+      select(getTestData),
+      select(getSpeedRequirementNotMet),
+      takeWhile(x => x),
+    ).subscribe(_ =>
+      this.store$.dispatch(new SetActivityCode(ActivityCodes.FAIL_PUBLIC_SAFETY)),
```

Comment:
This feels strange that the activity code is being set here. It should be set when leaving the test report